src/Service/Api/DataApi.php

```diff
@@ -307,25 +232,24 @@ public function saveEvent($event_name, $event_type = 'OTHER') {
    * @param $event_name
    *   The name of the event.
    *
-   * @throws AcquiaLiftDataConnectorException
+   * @throws DataApiException
    */
   public function deleteEvent($event_name) {
     // First get our Authorization header.
     $url = $this->generateEndpoint('events/' . $event_name);
     $auth_header = $this->getAuthHeader('DELETE', $url);
 
-    $request = new Request('DELETE', $url, array('Authorization' => $auth_header));
-    $response = $this->httpClient()->send($request);
+    $request = new Request('DELETE', $url, ['Authorization' => $auth_header]);
+    $response = $this->httpClient->send($request);
 
-    $vars = array('@eventname' => $event_name);
+    $vars = ['@eventname' => $event_name];
     $success_msg = SafeMarkup::format('The event @eventname was deleted from Acquia Lift Profiles', $vars);
     $fail_msg = SafeMarkup::format('Could not delete event @eventname from Acquia Lift Profiles', $vars);
     if ($response->getStatusCode() == 200) {
       $this->logger->info($success_msg);
+      return;
     }
-    else {
-      $this->logger->error($fail_msg);
-      throw new AcquiaLiftDataConnectorException($fail_msg);
-    }
+    $this->logger->error($fail_msg);
+    throw new DataApiException($fail_msg);
   }
 }
```

Comment:
DataApi class is not tested because it is not a part of any acceptance criteria.